src/Microsoft.AspNetCore.Razor.Evolution/IDirectiveDescriptorBuilder.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public interface IDirectiveDescriptorBuilder
+    {
+        IDirectiveDescriptorBuilder AddType();
+
+        IDirectiveDescriptorBuilder AddMember();
+
+        IDirectiveDescriptorBuilder AddString();
+
+        IDirectiveDescriptorBuilder AddLiteral(string literal, bool optional);
```

Comment:
My two ideas

1. Semicolons are always optional in directives (you don't have a choice)
2. Don't build `@addTagHelper` on top of extensible directives unless we want the "optional" part to become popular